pkg/vsphere/compute/placement/placement.go

```diff
@@ -17,15 +17,16 @@ package placement
 import (
 	"github.com/vmware/govmomi/object"
 	"github.com/vmware/vic/pkg/trace"
-	"github.com/vmware/vic/pkg/vsphere/vm"
+	"github.com/vmware/vic/pkg/vsphere/session"
 )
 
 // HostPlacementPolicy defines the interface for using metrics to decide the appropriate host
 // for a VM based on host metrics and VM provisioned resources.
 type HostPlacementPolicy interface {
 	// CheckHost checks whether or not the host a VM was created on is adequate for power-on.
-	CheckHost(trace.Operation, *vm.VirtualMachine) bool
+	CheckHost(trace.Operation, *session.Session) bool
```

Comment:
I'm not a fan of having session enshrined in the interface like this - we can have the implementor of the interface be the holder of a session if necessary.

As a second comment - how are we specifying which VM to check?

Reply:
Fixed the interface, but `CheckShost` is still a TODO for now.